backend/src/v1/data/repositories/wordRepository.ts

```diff
@@ -1,36 +1,80 @@
 import { Word } from "../models/models";
-
-interface WordRepository {
-    getWords(filterParams: {}): Word[];
-    getWord(id: string): Word;
-    addWord(data: AddWordData): void;
-    updateWord(word: Word): void;
-    deleteWord(id: string): void;
+import { db } from "../../../db/database";
+import { DeleteResult, InsertResult } from "kysely";
+export interface WordRepository {
+    getWords(languageCode?: string): Promise<Word[]>;
+    getWord(id: number): Promise<Word>;
+    insertWord(data: WordInsert): Promise<{ insertedRows: string; }>;
+    updateWord(data: WordUpdate): Promise<{ updatedRows: string; }>;
+    deleteWord(id: number): Promise<{ deletedRowsCount: string; }>;
 };
 
 const WordRepositoryImpl: WordRepository = {
-    getWords: function (filterParams: {}): Word[] {
-        throw new Error("Function not implemented.");
+    getWords: async (languageCode: string): Promise<Word[]> => {
+        let query = db.selectFrom("word").selectAll();
+        if (languageCode) {
+            query.where("language_code", "=", languageCode);
+        }
+        let result = await query.execute();
+        return result.map((entity) => {
```

Comment:
The code for these sorts of queries and transformations tends to be very boilerplatey. There are a few different ways to minimize this (see that talk that I linked to) but this is my favorite way to produce simple, well-typed queries: https://github.com/danvk/crudely-typed